ingester/src/query.rs

```diff
@@ -67,28 +67,40 @@ pub type Result<T, E = Error> = std::result::Result<T, E>;
 impl QueryableBatch {
     /// Initilaize a QueryableBatch
     pub fn new(table_name: &str, data: Vec<Arc<SnapshotBatch>>, deletes: Vec<Tombstone>) -> Self {
-        let mut delete_predicates = vec![];
-        for delete in &deletes {
-            let delete_predicate = Arc::new(
-                parse_delete_predicate(
-                    &delete.min_time.get().to_string(),
-                    &delete.max_time.get().to_string(),
-                    &delete.serialized_predicate,
-                )
-                .expect("Error building delete predicate"),
-            );
-
-            delete_predicates.push(delete_predicate);
-        }
-
+        let delete_predicates = Self::tombstones_to_delete_predicates(&deletes);
         Self {
             data,
-            deletes,
             delete_predicates,
             table_name: table_name.to_string(),
         }
     }
 
+    /// Add more tombstones
+    pub fn add_tombstones(&mut self, deletes: &mut Vec<Tombstone>) {
```

Comment:
It doesn't look like this function needs a mutable reference to a vec anymore, so I think this can be:

```suggestion
    pub fn add_tombstones(&mut self, deletes: &[Tombstone]) {
```

and then the call site can eliminate a `clone`, commenting there too...